src/main/java/org/springframework/data/couchbase/repository/support/SimpleReactiveCouchbaseRepository.java

```diff
@@ -0,0 +1,282 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.data.couchbase.repository.support;
+
+import java.io.Serializable;
+
+import com.couchbase.client.java.document.json.JsonArray;
+import com.couchbase.client.java.error.DocumentDoesNotExistException;
+import com.couchbase.client.java.view.ViewQuery;
+
+import org.reactivestreams.Publisher;
+import org.springframework.data.couchbase.core.ReactiveCouchbaseOperations;
+import org.springframework.data.couchbase.core.query.View;
+import org.springframework.data.couchbase.core.query.reactor.mapping.ReactiveViewResult;
+import org.springframework.data.couchbase.core.query.reactor.mapping.ReactiveViewRow;
+import org.springframework.data.couchbase.repository.ReactiveCouchbaseRepository;
+import org.springframework.data.couchbase.repository.query.CouchbaseEntityInformation;
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * Reactive repository base implementation for Couchbase.
+ *
+ * @author Subhashni Balakrishnan
+ * @since 3.0
+ */
+public class SimpleReactiveCouchbaseRepository<T, ID extends Serializable> implements ReactiveCouchbaseRepository<T, ID> {
+
+    /**
+     * Holds the reference to the {@link org.springframework.data.couchbase.core.ReactiveCouchbaseTemplate}.
+     */
+    private final ReactiveCouchbaseOperations operations;
+
+    /**
+     * Contains information about the entity being used in this repository.
+     */
+    private final CouchbaseEntityInformation<T, String> entityInformation;
+
+    /**
+     * Custom ViewMetadataProvider.
+     */
+    private ViewMetadataProvider viewMetadataProvider;
+
+    /**
+     * Create a new Repository.
+     *
+     * @param metadata the Metadata for the entity.
+     * @param operations the reference to the reactive template used.
+     */
+    public SimpleReactiveCouchbaseRepository(final CouchbaseEntityInformation<T, String> metadata,
```

Comment:
I have added the mapping here mapMono and mapFlux wrapper methods, it would be great to know your thoughts.